UI/DisplayLayoutScreen.cpp

```diff
@@ -0,0 +1,227 @@
+		g_Config.fSmallDisplayOffsetX = 0.5f;
+		g_Config.fSmallDisplayOffsetY = 0.5f;
+	}
+	RecreateViews();
+	return UI::EVENT_DONE;
+};
+
+
+void DisplayLayoutScreen::dialogFinished(const Screen *dialog, DialogResult result) {
+	RecreateViews();
+}
+
+void DisplayLayoutScreen::CreateViews() {
+	if (g_Config.bStretchToDisplay) {
+		// Shouldn't even be able to get here as the way into this dialog should be closed.
+		return;
+	}
+	const Bounds &bounds = screenManager()->getUIContext()->GetBounds();
+
+	local_dp_xres = bounds.w;
+	local_dp_yres = bounds.h;
+
+	using namespace UI;
+
+	I18NCategory *gr = GetI18NCategory("Graphics");
+
+	root_ = new AnchorLayout(new LayoutParams(FILL_PARENT, FILL_PARENT));
+
+	// Just visual boundaries of the screen, should be easier to use than imagination
+	float verticalBoundaryPositionL = local_dp_xres / 4.0f;
+	float verticalBoundaryPositionR = local_dp_xres - verticalBoundaryPositionL;
+	float horizontalBoundaryPositionL = local_dp_yres / 4.0f;
+	float horizontalBoundaryPositionR = local_dp_yres - horizontalBoundaryPositionL;
+	TabHolder *verticalBoundaryL = new TabHolder(ORIENT_VERTICAL, verticalBoundaryPositionL, new AnchorLayoutParams(0, 0, 0, 0, false));
+	TabHolder *verticalBoundaryR = new TabHolder(ORIENT_VERTICAL, verticalBoundaryPositionR, new AnchorLayoutParams(0, 0, 0, 0, false));
+	TabHolder *horizontalBoundaryL = new TabHolder(ORIENT_VERTICAL, verticalBoundaryPositionL * 2.0f, new AnchorLayoutParams(verticalBoundaryPositionL * 2.0f, horizontalBoundaryPositionL - 31.0f, 0, 0, true));
+	TabHolder *horizontalBoundaryR = new TabHolder(ORIENT_VERTICAL, verticalBoundaryPositionL * 2.0f, new AnchorLayoutParams(verticalBoundaryPositionL * 2.0f, horizontalBoundaryPositionR + 31.0f, 0, 0, true));
+	AnchorLayout *topBoundary = new AnchorLayout(new LayoutParams(FILL_PARENT, FILL_PARENT));
+	AnchorLayout *bottomBoundary = new AnchorLayout(new LayoutParams(FILL_PARENT, FILL_PARENT));
+	root_->Add(verticalBoundaryL);
+	root_->Add(verticalBoundaryR);
+	root_->Add(horizontalBoundaryL);
+	root_->Add(horizontalBoundaryR);
+	horizontalBoundaryL->AddTab("", topBoundary);
+	horizontalBoundaryR->AddTab("", bottomBoundary);
+
+	Choice *back = new Choice(gr->T("Back"), "", false, new AnchorLayoutParams(leftColumnWidth, WRAP_CONTENT, 10, NONE, NONE, 10));
+	static const char *zoomLevels[] = { "Auto", "1x", "2x", "3x", "4x", "5x", "6x", "7x", "8x", "9x", "10x" };
+	zoom_ = new PopupMultiChoice(&g_Config.iSmallDisplayZoom, gr->T("Zoom settings"), zoomLevels, 0, ARRAY_SIZE(zoomLevels), gr->GetName(), screenManager(), new AnchorLayoutParams(300, WRAP_CONTENT, verticalBoundaryPositionL * 2 - 150, NONE, NONE, 10));
+	zoom_->OnChoice.Handle(this, &DisplayLayoutScreen::OnZoomChange);
+
+	
+
+	mode_ = new ChoiceStrip(ORIENT_VERTICAL, new AnchorLayoutParams(leftColumnWidth, WRAP_CONTENT, 10, NONE, NONE, 158 + 64 + 10));
+	if (g_Config.iSmallDisplayZoom == 0) {
+		mode_->AddChoice(gr->T("Activated(Auto)"));
```

Comment:
Can we make this "Active (Auto)" or something (with the space)?  What do you think?

-[Unknown]